internal/engine/local/controller.go

```diff
@@ -152,6 +163,41 @@ func (c *Controller) start(ctx context.Context, spec ServeSpec, st store.RStore)
 
 	spanID := SpanIDForServeLog(proc.procNum)
 	proc.doneCh = c.execer.Start(ctx, spec.ServeCmd, logger.Get(ctx).Writer(logger.InfoLvl), statusCh, spanID)
+
+	if spec.ReadinessProbe != nil {
+		statusChangeFunc := processReadinessProbeUpdate(ctx, st, spec.ManifestName, proc.stillHasSameProcNum())
+		probeWorker, err := probeWorkerFromSpec(
+			c.probeManager,
+			spec.ReadinessProbe,
+			probe.WorkerOnStatusChange(statusChangeFunc))
+		if err != nil {
+			logger.Get(ctx).Warnf("Invalid readiness probe: %v", err)
+		} else {
+			proc.probeWorker = probeWorker
+			go proc.probeWorker.Run(ctx)
+		}
+	}
+}
+
+func processReadinessProbeUpdate(
+	ctx context.Context,
+	st store.RStore,
+	manifestName model.ManifestName,
+	stillHasSameProcNum func() bool) probe.StatusChangedFunc {
+	return func(status prober.Result, output string) {
+		if !stillHasSameProcNum() {
+			return
+		}
+		if status != prober.Success && output != "" {
+			w := logger.Get(ctx).Writer(logger.DebugLvl)
```

Comment:
`DebugLvl` vs `InfoLvl` is just arbitrary, I assumed it's lower priority than your app logs. In general, if the resource + probe are well-behaved, you don't _really_ care about the probe output even when it's failing, as that just means that your resource is starting up/restarting. For example, if you restart an app that takes 10-15 secs to initialize, an `HTTPGet` probe might fail with a timeout for a while there...but the failing state is already reflected via the resource no longer showing as ready.

In terms of logging behavior until first success, I totally agree and this is actually how it behaves. Mimicking K8s, the readiness state is initialized as failing, and the `StatusChangedFunc` only fires when the status _changes_, not on *every* probe invocation, and this is where Tilt is getting the probe output. It IS worth noting that means that if an app goes from success -> failure and stays in that state for a while, we'll only log the _first_ probe failure that caused the transition and not on every subsequent failing invocation.

Reply:
* VerboseLvl is more appropriate than DebugLvl here - DebugLvl is theoretically just for stuff of [interest only to tilt developers](https://github.com/tilt-dev/tilt/blob/ff147111c87493b4f0d1c42fbc10f8f2e0134cab/pkg/logger/logger.go#L24)
* FWIW, for context - some log level decisions are annoying because currently the log level is only configurable startup time, so if you have all your stuff running via Tilt and suddenly need to debug a probe, now you have to restart Tilt to change the log level, which might 1) take a long time since that can also involve restarting services or 2) make the issue go away since you're restarting everything, so you can no longer debug it.

I think you're right and keeping it off of InfoLvl is probably for the best (though, as above, I'd opt for VerboseLvl over DebugLvl)

Reply:
Gotcha - swapped to `VerboseLvl` in 2ab43437f8a062576bbeab679561f12cf93326c3.

Reply:
I am gonna talk to folks a bit design-wise to see if there's something that makes sense longer-term here; I feel like along the same lines as build vs runtime log separation, we might want more generic/similar approach for meta-Tilt logs like this (but that are still associated with a resource in some way).